### PR TITLE
feat: integrate show notification button option

### LINKED ISSUE #4847: [FR] Option to not show notifications icon, when there isnt any notification
### Description

I dont use notification, so its just cluttering my screen

### Impact

Users, who dont use notifications or at least not often

### Additional Context

I think its better option than not showing it at all, because what if?

### PR DESCRIPTION
**Pull request by migcarde**
### Feature Preview

close #4847 

https://github.com/AppFlowy-IO/AppFlowy/assets/29870701/e154a904-7b70-4a8c-bbbd-170cc3dba47e

#### PR Checklist

- [ ] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/workspace/application/settings/notifications/notification_settings_cubit.dart [hunk 1]
```diff
@@ -19,6 +22,16 @@ class NotificationSettingsCubit extends Cubit<NotificationSettingsState> {
...
+      getIt<KeyValueStorage>()
+          .getWithFormat(
+            KVKeys.showNotificationIcon,
+            (value) => bool.parse(value),
+          )
+          .then(
+            (value) => state.copyWith(
+              isShowNotificationsIconEnabled: value ?? true,
+            ),
+          );
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/application/settings/notifications/notification_settings_cubit.dart:34]: Await completion of both and then emit.

You don't even emit the initial value, as you only do `state.copyWith` here.

Comment (Xazin) [reply frontend/appflowy_flutter/lib/workspace/application/settings/notifications/notification_settings_cubit.dart:34]: This is how you should do it:
```
Future<void> _initialize() async {
  final settings = await UserSettingsBackendService().getNotificationSettings();
  _notificationSettings = notificationSettings;

  final showNotificationSetting = await getIt<KeyValueStorage>()
        .getWithFormat(KVKeys.showNotificationIcon, (v) => bool.parse(v));

  emit(
    state.copyWith(
      isNotificationsEnabled: _notificationSettings.notificationsEnabled,
      isNotificationsIconEnabled: showNotificationSetting,
    ),
  );
     
  _initCompleter.complete();
}

File: frontend/appflowy_flutter/lib/workspace/presentation/notifications/widgets/notification_button.dart [hunk 2]
```diff
@@ -24,23 +25,32 @@ class NotificationButton extends StatelessWidget {
...
+                        windowPadding: EdgeInsets.zero,
+                        margin: EdgeInsets.zero,
+                        popupBuilder: (_) =>
+                            NotificationDialog(views: views, mutex: mutex),
+                        child:
+                            _buildNotificationIcon(context, state.hasUnreads),
+                      ),
+                    ),
+                  )
+                : const SizedBox(),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/notifications/widgets/notification_button.dart:51]:
```suggestion
                : const SizedBox.shrink(),
```

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_notifications_view.dart [hunk 3]
```diff
@@ -34,6 +34,25 @@ class SettingsNotificationsView extends StatelessWidget {
...
+              trailing: [
+                Switch(
+                  value: state.isShowNotificationsIconEnabled,
+                  splashRadius: 0,
+                  activeColor: Theme.of(context).colorScheme.primary,
+                  onChanged: (value) {
+                    context
+                        .read<NotificationSettingsCubit>()
+                        .toogleShowNotificationIconEnabled();
+                  },
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_notifications_view.dart:52]:
```suggestion
                  onChanged: (_) =>
                      context.read<NotificationSettingsCubit>()
                        .toogleShowNotificationIconEnabled(),
```

File: frontend/appflowy_flutter/lib/workspace/application/settings/notifications/notification_settings_cubit.dart (hunk 1 above)
Comment (migcarde) [reply frontend/appflowy_flutter/lib/workspace/application/settings/notifications/notification_settings_cubit.dart]: Applied proposal changes, follow [this link](https://github.com/AppFlowy-IO/AppFlowy/pull/5302/commits/b4acc1951f0434ddfac03c4c11383c486fbbd126) to see it

File: frontend/appflowy_flutter/lib/workspace/application/settings/notifications/notification_settings_cubit.dart [hunk 4]
```diff
@@ -10,23 +13,31 @@ part 'notification_settings_cubit.freezed.dart';
...
   }
 
   final Completer<void> _initCompleter = Completer();
 
   late final NotificationSettingsPB _notificationSettings;
 
+  Future<void> _initialize() async {
+    final settings =
+        await UserSettingsBackendService().getNotificationSettings();
+    _notificationSettings = settings;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/application/settings/notifications/notification_settings_cubit.dart:26]: I guess it was my bad, we can actually just do
```suggestion
    _notificationSettings =
        await UserSettingsBackendService().getNotificationSettings();
```

**Review by Xazin — APPROVED** [review]

**Comment by Xazin** [comment]
Seems there are some issues due to merge.

You might have to rename the usage of `FlowySettingListTile` to `SettingListTile`.

**Comment by migcarde** [comment]
> Seems there are some issues due to merge.
> 
> You might have to rename the usage of `FlowySettingListTile` to `SettingListTile`.

Resolved naming with this Widget.

**Pull request merged**